### PR TITLE
Jetpack Connect: Display a detailed invalid password error during signup

### PR DESCRIPTION
**Pull request by tyxla**
This PR updates the Jetpack Connect signup process to display a detailed error message in the case where an invalid password is specified. 

The PR suggests the following:
* Recognizes invalid password errors and displays them in detail in the error notice.
* Properly clears the signup form submitting state on that kind of error, so we can submit the form again.
* Adds an ID to the invalid password error notice, so we don't duplicate them if the user tries more than once with an invalid password.

Note: it seems that the `SignupForm` accepts a `submitting` prop, but also stores `submitting` in its component state. Those two aren't in sync, which can cause all kind of problems. The PR solves the case for errors, but doesn't fix any other pre-existing race condition problems that might occur. A long-term solution would be to refactor `SignupForm` to not have both component state and a prop, but this is material for another PR.

**Testing:**
For detailed testing instructions, head over to D18476-code.

### CODE REVIEW

**Comment by ockham** [comment]
Following instructions in D18476-code, I'm still getting the 'generic' error message :slightly_frowning_face: 

image

This is with a `me+dummy` A8c email address. Signup without a backslash in the password works however.

(I did restart my browser after sandboxing...)

**Comment by ockham** [comment]
Aside, shouldn't we check for forbidden characters on the frontend side, i.e. while entering the password/when clicking the submit button?

File: client/jetpack-connect/signup.js [hunk 1]
```diff
@@ -164,6 +165,10 @@ export class JetpackSignup extends Component {
 			} );
 			return;
 		}
+		if ( error && error.error && 'password_invalid' === error.error ) {
```
Comment (ockham) [inline client/jetpack-connect/signup.js]: What about `if ( get(  error, [ 'error' ] ) === 'password_invalid' ) {`?

**Comment by tyxla** [comment]
> Aside, shouldn't we check for forbidden characters on the frontend side, i.e. while entering the password/when clicking the submit button?

Good point. I totally agree, and TBH I considered it before pushing this fix, but when I saw that it'll require a complete overhaul of the validation mechanism, I decided it should be done in another PR.

File: client/jetpack-connect/signup.js (hunk 1 above)
Comment (tyxla) [reply client/jetpack-connect/signup.js]: Nice 👍 Updated in https://github.com/Automattic/wp-calypso/pull/27285/commits/4e897f5d9fe95b1be581a18570f9ffa7a47f7ee9.

**Comment by ebinnion** [comment]
Fixes #27197.

**Pull request merged**